packages/headless/src/modules/order/services/customer.service.ts

```diff
@@ -32,6 +33,42 @@ export class OrderCustomerService {
     );
   }
 
+  async reOrder(userId: string, body: ReOrderQuery): Promise<IServiceResponse<CartResponse>>{
+    let { ignoreInvalidItems, overWriteCart, orderId } = body; 
+
+    const prevOrder = await this.orderRepository.findOrder({orderId, userId});
+    if(!prevOrder) return errorResponse( ErrorMessageReOrder.INVALID_ID,null,HttpStatus.BAD_REQUEST );
+
+    const prevProducts = prevOrder.products;
+    const productIds = prevProducts.map( item => item.productId);
+    let order = prevProducts.map(product => { return {productId: product.productId, quantity: product.quantity}});
+
+    const availableProductIds = await this.orderRepository.getAvailableProducts(productIds);
+
+    if(availableProductIds.length === 0)return errorResponse( ErrorMessageReOrder.ALL_ITEMS_INVALID,{products: prevProducts.map(item => item.name)},HttpStatus.BAD_REQUEST );//send which products are not available
+    else {
+          const unavailableProducts = prevProducts.filter(product => !availableProductIds.find(item => item.id === product.productId) && product.name);
+       
+          if(ignoreInvalidItems === false) return errorResponse( ErrorMessageReOrder.INVALID_ITEMS, {products: unavailableProducts.map(item => item.name)}, HttpStatus.BAD_REQUEST );
+          
+          order = order.filter(product => availableProductIds.find(item => item.id === product.productId));    
+        }
+
+        const cart = await this.orderRepository.getCart(userId);
+        if (cart.items.length !== 0){
+            if(overWriteCart === true){
+                const deleteCart = await this.orderRepository.clearCart(userId);
+                if(!deleteCart) return errorResponse( ErrorMessageReOrder.CANNOT_CLEAR_CART, null, HttpStatus.BAD_REQUEST );
+
+            }else return errorResponse( ErrorMessageReOrder.OVERWRITE_CART, null, HttpStatus.BAD_REQUEST );
```

Comment:
return those item names that are already in cart. and also return meaningful message so that front end can identify that the cart has some items